From f3cc0d34355f1a8f244007048a38c2112b7c15f6 Mon Sep 17 00:00:00 2001
Message-Id: <f3cc0d34355f1a8f244007048a38c2112b7c15f6.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Akash Goel <akash.goel@intel.com>
Date: Tue, 10 Jun 2014 15:47:22 +0530
Subject: [PATCH 111/312] FOR_UPSTREAM [VPG]: drm/i915: Set min freq to Rpe on
 VLV

On VLV, Max freq at Vmin is Rpe, so for better perfomance lets operate
between Rpe and Rp0. Not much power savings in going below Rpe to Rpn.

Issue: VIZ-3934
Change-Id: I8c3a2a1e0d286c853ac33bddddbe18ba9d6200cf
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Akash Goel <akash.goel@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 2229b93..3e718df 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4251,8 +4251,9 @@ static void valleyview_init_gt_powersave(struct drm_device *dev)
 	if (dev_priv->rps.max_freq_softlimit == 0)
 		dev_priv->rps.max_freq_softlimit = dev_priv->rps.max_freq;
 
+	/* on VLV, RPe Freq at Vmin. So set min_delay = RPe_delay*/
 	if (dev_priv->rps.min_freq_softlimit == 0)
-		dev_priv->rps.min_freq_softlimit = dev_priv->rps.min_freq;
+		dev_priv->rps.min_freq_softlimit = dev_priv->rps.efficient_freq;
 
 	mutex_unlock(&dev_priv->rps.hw_lock);
 }
-- 
1.7.9.5

